docarray/document/mixins/multimodal.py

```diff
@@ -139,3 +139,21 @@ def _from_obj(cls, obj, obj_type, field) -> typing.Tuple['Document', AttributeTy
         else:
             raise ValueError(f'Unsupported type annotation')
         return doc, attribute_type
+
+    def __getattr__(self, attr):
+        try:
+            data = super().__getattribute__('_data')
+            has_data = bool(data)
+        except AttributeError:
+            has_data = False
+
+        has_metadata = has_data and getattr(self._data, '_metadata') is not None
+        if (
+            has_metadata
+            and self.is_multimodal
+            and attr in self._metadata['multi_modal_schema']
+        ):
+            mm_attr_da = self.get_multi_modal_attribute(attr)
+            return mm_attr_da if len(mm_attr_da) > 1 else mm_attr_da[0]
+        else:
+            raise AttributeError(f'{self.__class__.__name__} has no attribute {attr}')
```

Comment:
I don't think we should refer to the class here but rather to the object. An Python object can have an attribute without it being define in the class ( we can add attribute on the fly after object instantiation)

Reply:
Are you referring to the error message? i don't think that the object itself has a `__name__`, could you write a code change suggestion showing what you would do?